flytekit/taskplugins/sqlite/task.py

```diff
@@ -0,0 +1,113 @@
+    to configure a path to a static sqlite3 compatible database file that is available either
+      - within the container
+      - or from a publicly downloadable source
+
+    If the Mode is marked dynamic, then a variable called ``sqlite: FlyteFile[".sqlite"]`` will be automatically added
+    to the input interface of the task. Failing to supply this at runtime will raise an Assertion
+
+        Args:
+            sqlite_db_mode: STATIC should be use for constant locations of file, dynamic to allow dynamically provided
+                            sqlite3 file locations. The file will have to comply with FlyteFile[.db|.zip|.tar.gz|.bz2]
+                            format
+            static_file_path: default FlyteFile that will be downloaded on execute
+            compressed: Boolean that indicates if the given file is a compressed archive
+    """
+
+    class Mode(Enum):
+        STATIC = "static"
+        """
+        STATIC refers to a constant, statically configured reference.
+        """
+        DYNAMIC = "dynamic"
+        """
+        Dynamic refers to SQLite files will be dynamically injected into the task.
+        """
+
+    sqlite_db_mode: Mode = Mode.DYNAMIC
+    static_file_uri: typing.Optional[str] = None
+    compressed: bool = False
+
+
+class SQLite3Task(SQLTask[SQLite3Config]):
+    """
+    Makes it possible to run client side SQLite3 queries that optionally return a FlyteSchema object
+
+    TODO: How should we use pre-built containers for running portable tasks like this. Should this always be a
+          referenced task type?
+    """
+
+    _SQLITE_INPUT_FILE = "sqlite"
+
+    def __init__(
+        self,
+        name: str,
+        query_template: str,
+        metadata: typing.Optional[_task_models.TaskMetadata] = None,
+        inputs: typing.Optional[typing.Dict[str, typing.Type]] = None,
+        task_config: typing.Optional[SQLite3Config] = None,
+        output_schema_type: typing.Optional[typing.Type[FlyteSchema]] = None,
+        *args,
+        **kwargs,
+    ):
+        if not inputs:
+            inputs = {}
+        if task_config is not None and task_config.sqlite_db_mode == SQLite3Config.Mode.STATIC:
+            if task_config.static_file_uri is None:
+                raise ValueError("SQLite DB mode configured to be static, but, the static_file_path was not specified.")
+        else:
+            if self._SQLITE_INPUT_FILE in inputs:
+                raise ValueError(f"Input name `{self._SQLITE_INPUT_FILE}` is reserved as an auto-injected input")
+            inputs[self._SQLITE_INPUT_FILE] = FlyteFile
+        if output_schema_type is None:
+            output_schema_type = FlyteSchema
+        outputs = kwtypes(results=output_schema_type)
+        if metadata is None:
+            metadata = md()
+        super().__init__(
+            name, metadata, query_template, inputs, outputs=outputs, task_config=task_config, *args, **kwargs
```

Comment:
is task_type unchanged from the parent class?

Reply:
good catch, it does not really matter, as this is a python task, but has a lot of benefits for the UI

Reply:
i actually fixed it in my next PR, if you guys are ok, I will merge that PR into this one?